web/ui/mantine-ui/src/pages/RulesPage.tsx

```diff
@@ -68,79 +142,91 @@ export default function RulesPage() {
     }
   }, [effectiveActivePage, activePage, setActivePage]);
 
-  return (
-    <Stack mt="xs">
-      {data.data.groups.length === 0 && (
-        <Alert title="No rule groups" icon={<IconInfoCircle />}>
-          No rule groups configured.
-        </Alert>
-      )}
-      <Pagination
-        total={totalPageCount}
-        value={effectiveActivePage}
-        onChange={setActivePage}
-        hideWithOnePage
-      />
-      {data.data.groups
-        .slice(
-          (effectiveActivePage - 1) * ruleGroupsPerPage,
-          effectiveActivePage * ruleGroupsPerPage
-        )
-        .map((g) => (
-          <Card
-            shadow="xs"
-            withBorder
-            p="md"
-            mb="md"
-            key={`${g.file}-${g.name}`}
-          >
-            <Group mb="sm" justify="space-between">
-              <Group align="baseline">
-                <Text fz="xl" fw={600} c="var(--mantine-primary-color-filled)">
-                  {g.name}
-                </Text>
-                <Text fz="sm" c="gray.6">
-                  {g.file}
-                </Text>
-              </Group>
-              <Group>
-                <Tooltip label="Last group evaluation" withArrow>
-                  <Badge
-                    variant="light"
-                    className={badgeClasses.statsBadge}
-                    styles={{ label: { textTransform: "none" } }}
-                    leftSection={<IconRefresh style={badgeIconStyle} />}
-                  >
-                    last run {humanizeDurationRelative(g.lastEvaluation, now())}
-                  </Badge>
-                </Tooltip>
-                <Tooltip label="Duration of last group evaluation" withArrow>
-                  <Badge
-                    variant="light"
-                    className={badgeClasses.statsBadge}
-                    styles={{ label: { textTransform: "none" } }}
-                    leftSection={<IconHourglass style={badgeIconStyle} />}
-                  >
-                    took {humanizeDuration(parseFloat(g.evaluationTime) * 1000)}
-                  </Badge>
-                </Tooltip>
-                <Tooltip label="Group evaluation interval" withArrow>
-                  <Badge
-                    variant="transparent"
-                    className={badgeClasses.statsBadge}
-                    styles={{ label: { textTransform: "none" } }}
-                    leftSection={<IconRepeat style={badgeIconStyle} />}
-                  >
-                    every {humanizeDuration(parseFloat(g.interval) * 1000)}{" "}
-                  </Badge>
-                </Tooltip>
-              </Group>
+  const currentPageGroups = useMemo(
+    () =>
+      shownGroups.slice(
+        (effectiveActivePage - 1) * ruleGroupsPerPage,
+        effectiveActivePage * ruleGroupsPerPage
+      ),
+    [shownGroups, effectiveActivePage, ruleGroupsPerPage]
+  );
+
+  // We memoize the actual rendering of the page items to avoid re-rendering
```

Comment:
```suggestion
  // We memorize the actual rendering of the page items to avoid re-rendering
```

Reply:
It's actually meant to be "memoize" :) https://en.wikipedia.org/wiki/Memoization

Reply:
oh ok. I didn't know this word. Thank you for enriching my knowledge 🙏